src/Core/src/Platform/Android/Resources/values/styles.xml

```diff
@@ -44,6 +45,10 @@
     <style name="AppTheme" parent="Maui.MainTheme" />
     <style name="AppTheme.NoActionBar" parent="Maui.MainTheme.NoActionBar" />
 
+    <style name="ToolbarStyle" parent="@style/Widget.AppCompat.Toolbar">
```

Comment:
Maybe this should be called MauiToolbarStyle to be consistent.

But also, how does this work with dark and light themes?

And then the question comes in if the default android style is white, where is the black coming from... @PureWeen is there maybe some color property in Maui that is null and as a result getting converted into the fallback black?

Reply:
Yea, I'd be curious to figure out what's setting this color. Is this something in our code? or is this how the default android style works? We're using `Theme.MaterialComponents.DayNight.DarkActionBar` as the base theme in Android. Are the defaults of that theme bad for high contrast requirements?

The original issue for this is using `NavigationPage`. With `NavigationPage` the toolbar colors aren't settable through any xplat APIs which is why the default MAUI template looks fine because `Shell` does let you set those colors.

Let's first confirm that what you see with `NavigationPage` are the defaults of Theme.MaterialComponents.DayNight.DarkActionBar` and not just us doing something wrong in code.

Reply:
It seems that default navigation toolbar color is set here: https://github.com/dotnet/maui/blob/f95920d886315eeea7a5737a07fcb5379ad5be8e/src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs#L168

Reply:
Renamed `ToolbarStyle` to `MauiToolbarStyle`

Reply:
In dark mode toolbar text color is white, so no problems there.

Reply:
> That doesn't give much information. The PR you wrote is about fixing bad contrast between two colors, you're only mentioning one color here.

Background is the same color (dark) in both modes.